### PR TITLE
Feature/newbackend

### PR DESCRIPTION
**Pull request by Letaguator**

### CODE REVIEW

**Comment by petterroea** [comment]
You committed the `.vs` folder, this has to be removed

**Comment by Letaguator** [comment]
> You committed the `.vs` folder, this has to be removed

I did not realize, should not that be in the .gitignore?

**Comment by petterroea** [comment]
It is, on my branch

**Comment by petterroea** [comment]
I suggest you delete the `.vs` folder and commit that

**Comment by Letaguator** [comment]
Sounds good, will do that!

**Comment by petterroea** [comment]
Please also delete the `.vs` folder and commit this - gitignore only blocks further changes from being detected, it does not blacklist the folder entirely

**Comment by Letaguator** [comment]
I am fully aware of it :) I just added it to the .gitignore by using GitHub, seemed cumbersome to delete the files from GitHub directly too so decided to do that locally.

**Comment by petterroea** [comment]
Aight! HMU when you are ready, and i'l give it a proper look

File: backend/Domain/Models/AuthModel.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+
+namespace OpenData.Domain.Models
+{
+    /// <summary>
+    /// Used for loggin user in.
+    /// </summary>
+    public class AuthModel
```
Comment (petterroea) [inline backend/Domain/Models/AuthModel.cs]: Shouldn't this be a Resource?

File: backend/Domain/Models/NewUserModel.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+
+namespace OpenData.Domain.Models
+{
+    /// <summary>
+    /// Form for creating new user
+    /// </summary>
+    public class NewUserModel
```
Comment (petterroea) [inline backend/Domain/Models/NewUserModel.cs]: This as well looks like a Resource. The models folder should contain things that exist in databases, these classes seem to just be for validation

File: backend/Domain/Models/User.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        [Description("admin")]
+        Admin
+    }
+
+    public class User
+	{
+		public string Mail { get; set; }
+		public string Password { get; set; }
+		public string PasswordSalt { get; set; }
+        public UserType UserType { get; set; } = UserType.Standard;
```
Comment (petterroea) [inline backend/Domain/Models/User.cs]: Fix tab

File: backend/Helpers/Hasher.cs [hunk 4]
```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Security.Cryptography;
+using System.Text;
+using Microsoft.AspNetCore.Cryptography.KeyDerivation;
+
+namespace OpenData.Helpers
+{
+    public static class Hasher
```
Comment (petterroea) [inline backend/Helpers/Hasher.cs]: This should not be a static class, you should use dependency injection for this.

ISecurityProvider, and then SecurityProvider.

File: backend/Helpers/SafeTransportMethods.cs [hunk 5]
```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using OpenData.Domain.Models;
+
+namespace OpenData.Helpers
+{
+    public static class SafeTransportMethods
```
Comment (petterroea) [inline backend/Helpers/SafeTransportMethods.cs]: We can use the automapper for this instead

File: backend/Helpers/Validator.cs [hunk 6]
```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Text.RegularExpressions;
+using System.Net.Mail;
+
+namespace OpenData.Helpers
+{
+    public static class Validator
```
Comment (petterroea) [inline backend/Helpers/Validator.cs]: This also should be either dependency injected, or it should be a function on user that throws an exception if you try to create it with an invalid email

File: backend/Properties/launchSettings.json [hunk 7]
```diff
@@ -1,8 +1,8 @@
-﻿{
+{
```
Comment (petterroea) [inline backend/Properties/launchSettings.json]: This file is not changed, your IDE has been doing autoformatting. This is a bad thing

File: backend/appsettings.json [hunk 8]
```diff
@@ -7,5 +7,8 @@
   "AllowedHosts": "*",
   "ConnectionStrings": {
     "DefaultConnection": "server=mysql;port=3306;database=OpenData;user=OpenData;password=OpenData"
+  },
+  "AppSettings": {
+    "Secret": "THIS IS MATHIAS' SUPER SECRET AMAZING STRING! (#$ERFDCVBHJKIOIUYT6346347RFDC345FV)"
```
Comment (petterroea) [inline backend/appsettings.json]: :thinking:

File: backend/backend.csproj [hunk 9]
```diff
@@ -10,9 +10,14 @@
...
-      <Version>3.1.0</Version>
+    <PackageReference Include="Pomelo.EntityFrameworkCore.MySql" Version="3.1.0">
     </PackageReference>
+    <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="5.6.0" />
+    <PackageReference Include="Microsoft.AspNetCore.Authentication.JwtBearer" Version="3.1.1" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.Analyzers" Version="3.1.1" />
   </ItemGroup>
 
+  <ItemGroup>
+    <Folder Include="Helpers\" />
```
Comment (petterroea) [inline backend/backend.csproj]: What is this?

File: backend/appsettings.json (hunk 8 above)
Comment (fipa98) [reply backend/appsettings.json]: 🤔

File: backend/Domain/Models/User.cs (hunk 3 above)
Comment (Letaguator) [reply backend/Domain/Models/User.cs]: image

Comment (Letaguator) [reply backend/Domain/Models/User.cs]: Very weird, cannot reproduce the tab error.

File: backend/backend.csproj (hunk 9 above)
Comment (Letaguator) [reply backend/backend.csproj]: Are you referring to line 21?

File: backend/Properties/launchSettings.json (hunk 7 above)
Comment (Letaguator) [reply backend/Properties/launchSettings.json]: Are you sure it is just auto formatting, line 24 seems to have been removed by the IDE?
And as an addendum a linter would have caught the unnecessary formatting?

File: backend/Domain/Models/NewUserModel.cs (hunk 2 above)
Comment (Letaguator) [reply backend/Domain/Models/NewUserModel.cs]: Resolved in newer commit.

File: backend/Domain/Models/AuthModel.cs (hunk 1 above)
Comment (Letaguator) [reply backend/Domain/Models/AuthModel.cs]: Resolved in newer commit.

File: backend/Helpers/SafeTransportMethods.cs (hunk 5 above)
Comment (Letaguator) [reply backend/Helpers/SafeTransportMethods.cs]: Resolved in newer commit.

File: backend/Helpers/Validator.cs (hunk 6 above)
Comment (Letaguator) [reply backend/Helpers/Validator.cs]: Resolved in newer commit.

File: backend/Resources/PrivateSafeUserResource.cs [hunk 10]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using OpenData.Domain.Models;
+
+namespace OpenData.Resources
+{
+    /// <summary>
+    /// Used for safely transporting userdata without passwords, salts and other classified information.
+    /// </summary>
+    public class PrivateSafeUserResource
```
Comment (petterroea) [inline backend/Resources/PrivateSafeUserResource.cs]: good stuff!

**Pull request merged**